mmdet/datasets/transforms/loading.py

```diff
@@ -303,22 +303,47 @@ def _poly2mask(self, mask_ann: Union[list, dict], img_h: int,
         mask = maskUtils.decode(rle)
         return mask
 
-    def process_polygons(self, polygons: List[list]) -> List[np.ndarray]:
-        """Convert polygons to list of ndarray and filter invalid polygons.
+    def _process_masks(self, results: dict) -> list:
+        """Process gt_masks and filter invalid polygons.
 
         Args:
-            polygons (list[list]): Polygons of one instance.
+            results (dict): Result dict from :obj:``mmcv.BaseDataset``.
 
         Returns:
-            list[np.ndarray]: Processed polygons.
+            list: Processed gt_masks.
         """
-
-        polygons = [np.array(p) for p in polygons]
-        valid_polygons = []
-        for polygon in polygons:
-            if len(polygon) % 2 == 0 and len(polygon) >= 6:
-                valid_polygons.append(polygon)
-        return valid_polygons
+        gt_masks = []
+        for instance in results['instances']:
+            gt_mask = instance['mask']
+            # If the annotation of segmentation mask is invalid,
+            # ignore the whole instance.
+            if isinstance(gt_mask, list):
+                gt_mask = [
+                    np.array(polygon) for polygon in gt_mask
+                    if len(polygon) % 2 == 0 and len(polygon) >= 6
+                ]
+                if len(gt_mask) == 0:
+                    # ignore this instance and set gt_mask to a fake mask
+                    instance['ignore_flag'] = 1
```

Comment:
Is this logic correct? Because _load_bboxes has been converted to gt_ignore_flags. Is it reasonable to set ignore_flag at this time?

Reply:
re-convert gt_ignore_flaags in _load_mask